echo_server.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import unicode_literals
+import socket
+
+
+server_socket = socket.socket(
+    socket.AF_INET,
+    socket.SOCK_STREAM,
+    socket.IPPROTO_IP)
+
+buffer_size = 32
+
+server_socket.bind(('127.0.0.1', 50000))
+server_socket.listen(1)
+
+
+try:
+    while True:
+        msg = 'Echo: "'
```

Comment:
By adding this to the value you send back to the client, you are no longer echoing.  You are altering the message in transit.  Don't.